src/ucs/sys/topo.c

```diff
@@ -130,7 +136,7 @@ ucs_status_t ucs_topo_get_distance(ucs_sys_device_t device1,
     ucs_topo_get_path_with_bus(bus2, path2);
 
     path_distance = ucs_path_calc_distance(path1, path2);
-    if (path_distance == UINT_MAX) {
+    if (path_distance < 0) {
```

Comment:
if err, convert path_distance to ucs_status_t an return it as the status, not IO_ERROR